be/src/olap/single_replica_compaction.cpp

```diff
@@ -396,8 +398,22 @@ Status SingleReplicaCompaction::_download_files(DataDir* data_dir,
     uint64_t total_file_size = 0;
     MonotonicStopWatch watch;
     watch.start();
+    CURL* curl = curl_easy_init();
```

Comment:
use unique_ptr with custom delete function to avoid memory leak if return before curl_easy_cleanup